chromadb/api/client.py

```diff
@@ -56,13 +57,28 @@ def __init__(
         super().__init__(settings=settings)
         self.tenant = tenant
         self.database = database
-        # Create an admin client for verifying that databases and tenants exist
-        self._admin_client = AdminClient.from_system(self._system)
-        self._validate_tenant_database(tenant=tenant, database=database)
 
         # Get the root system component we want to interact with
         self._server = self._system.instance(ServerAPI)
 
+        user_identity = self.resolve_tenant_and_databases()
+
+        maybe_tenant, maybe_database = SharedSystemClient.maybe_set_tenant_and_database(
```

Comment:
What happens if the user supplied tenant or database doesn't match?

Reply:
They are only overridden if they are (not provided or the default values) and we can definitely determine what they should be based on the API key (i.e. a specific tenant and only one specific database exists). Thus, if they don't match but the API key is correct, then the request will fail authorization since it's a mismatch of resources.

Reply:
I don't see where that would happen?

User does: chromadb.Client(tenant=FOO, database=BLAH)
Code does: self.get_user_identity() -> Returns UserIdentity(tenant=NOT_FOO, database=SOMETHING)
Code does: maybe_set_tenant_and_database will just set tenant and database to NOT_FOO, SOMETHING

I'm probably missing something.

Reply:
Right and I am saying should we make the client init fail, under a case like this, rather than deferring the failure to the first request the user will make?

Reply:
Ah, got it. Updated to throw an exception in this error case and moved resolution to the auth library under a new utils section (I thought in your other comments were suggesting to throw it into the server auth provider).